asciidoc/cypher-workflow.adoc

```diff
@@ -0,0 +1,804 @@
+When using a single instance, all transactions will be passed to that one server.
+
+Routing Cypher by identifying reads and writes can improve the utilization of available cluster resources.
+Since read servers are typically more plentiful than write servers, it is beneficial to direct read traffic to read servers instead of the write server.
+Doing so helps in keeping write servers available for write transactions.
+
+Access mode is generally specified by the method used to call the transaction function.
+Session classes provide a method for calling reads and another for writes.
+
+As a fallback for auto-commit and unmanaged transactions, a default access mode can also be provided at session level.
+This is only used in cases when the access mode cannot otherwise be specified, and will be overridden should a transaction function be used within that session.
+
+[NOTE]
+The driver does not parse Cypher and therefore cannot automatically determine whether a transaction is intended to carry out read or write operations.
+As a result, a write transaction tagged as a read will still be sent to a read server, but will fail on execution.
+
+[.tabbed-example]
+.Read-write transaction
+====
+[.include-with-dotnet]
+======
+[source, csharp]
+----
+include::{dotnet-examples}/Examples.cs[tags=read-write-transaction]
+----
+======
+
+// [.include-with-go]
+// ======
+// [source, go]
+// ----
+// include::{go-examples}/examples_test.go[tags=read-write-transaction]
+// ----
+// ======
+
+[.include-with-java]
+======
+[source, java]
+----
+include::{java-examples}/ReadWriteTransactionExample.java[tags=read-write-transaction]
+----
+======
+
+[.include-with-javascript]
+======
+[source, javascript]
+----
+include::{javascript-examples}/examples.test.js[tags=read-write-transaction]
+----
+======
+
+// [.include-with-python]
+// ======
+// [source, python]
+// ----
+// include::{python-examples}/read_write_transaction_example.py[tags=read-write-transaction]
+// ----
+// ======
+====
+
+[[driver-databases]]
+== Databases and execution context
+
+Neo4j 4.0 introduces the ability to work with multiple databases within the same DBMS.
+For a single instance, this is limited to one user database, plus the `system` database.
+From a driver API perspective, a database can be selected on session construction, and is used as an execution context for the transactions within that session.
+It is not currently possible to execute a transaction across multiple databases.
+
+In a multi-database environment, the server tags one database as default.
+This is selected whenever a session is created without naming a particular database as default.
+
+In an environment with only a single database, that database is always the default.
+
+For more information about the multi-database feature, refer to the <<cypher-manual#cypher-manual, Cypher Manual>> which has a full breakdown of the Neo4j data storage hierarchy.
```

Comment:
It has been decided that there is no feature called "the multi-database feature". Instead, it could be worded something similar to this:
"For more information about managing multiple databases within the same DBMS, refer to...".

(If we want to link directly to the administrative commands for multiple databases, the link is cypher-manual#administration-databases, but that doesn't explain the concepts. The section that explains the concepts is cypher-manual#neo4j-databases-graphs)

Reply:
New wording seems much better :)

I'm happy to look at rewording the info on multidb. This was difficult to write, and maybe there's a better way to approach it.

Reply:
Updated